### PR TITLE
Improve matching for artist and mapper search

### LINKED ISSUE #7333: Artist and creator advanced search improvements
It currently works in a rather limited way. Here are list of things need to be added:

- prefix and keyword index like title for `artist`, `artist_unicode`, and `creator` fields (I noticed `title_unicode` is also missing those index)
- `artist_unicode` needs to be counted in for artist search (similar to general search)
- both artist and creator search should allow term/exact match

(and maybe more things I missed/forgot)

### PR DESCRIPTION
**Pull request by nanaya**
Resolves #7333.

### CODE REVIEW

File: app/Libraries/Search/BeatmapsetSearch.php [hunk 1]
```diff
@@ -33,7 +33,19 @@ public function __construct(?BeatmapsetSearchParams $params = null)
...
     {
-        static $partialMatchFields = ['artist', 'artist.*', 'artist_unicode', 'creator', 'title', 'title.raw', 'title.*', 'title_unicode', 'tags^0.5'];
+        static $partialMatchFields = [
+            'artist',
+            'artist.*',
+            'artist_unicode',
+            'artist_unicode.*',
+            'creator',
+            'title',
+            'title.raw',
```
Comment (nanaya) [inline app/Libraries/Search/BeatmapsetSearch.php:43]: not sure if useful considering there's also `title.*`

**Review by notbakaneko — CHANGES_REQUESTED** [review]
Is the query `artist:"the best"` supposed to search for 
- `the best`
- `the` or `best` or `the best`

File: app/Libraries/Search/BeatmapsetSearch.php (hunk 1 above)
Comment (notbakaneko) [reply app/Libraries/Search/BeatmapsetSearch.php:43]: Should be fine to remove 🤔

**Comment by peppy** [comment]
I would hope it searches for `the best` only.

**Comment by notbakaneko** [comment]
I forgot to mention but should `artist:"the best"` match `the best one`?

**Comment by peppy** [comment]
Yep

**Review by notbakaneko — APPROVED** [review]

**Pull request merged**